Gems/ROS2/Code/Source/Lidar/ClassSegmentationConfigurationComponent.cpp

```diff
@@ -0,0 +1,133 @@
+                    ->Attribute(
+                        AZ::Edit::Attributes::ChangeValidate, &ClassSegmentationConfigurationComponent::ValidateSegmentationClasses);
+            }
+        }
+    }
+
+    AZ::Color ClassSegmentationConfigurationComponent::GetClassColor(uint8_t classId) const
+    {
+        auto it = m_classIdToColor.find(classId);
+        if (it == m_classIdToColor.end())
+        {
+            return AZ::Colors::White;
+        }
+
+        return it->second;
+    }
+
+    AZStd::optional<uint8_t> ClassSegmentationConfigurationComponent::GetClassIdForTag(LmbrCentral::Tag tag) const
+    {
+        auto it = m_tagToClassId.find(tag);
+        if (it == m_tagToClassId.end())
+        {
+            return AZStd::nullopt;
+        }
+
+        return it->second;
+    }
+
+    const SegmentationClassConfigList& ClassSegmentationConfigurationComponent::GetClassConfigList() const
+    {
+        return m_segmentationClasses;
+    }
+
+    void ClassSegmentationConfigurationComponent::Activate()
+    {
+        ConstructSegmentationClassMaps();
+
+        if (!ClassSegmentationInterface::Get())
+        {
+            ClassSegmentationInterface::Register(this);
+        }
+
+        ClassSegmentationNotificationBus::Broadcast(&ClassSegmentationNotifications::OnSegmentationClassesReady);
+    }
+
+    void ClassSegmentationConfigurationComponent::Deactivate()
+    {
+        if (ClassSegmentationInterface::Get() == this)
+        {
+            ClassSegmentationInterface::Unregister(this);
+        }
+    }
+
+    AZ::Outcome<void, AZStd::string> ClassSegmentationConfigurationComponent::ValidateSegmentationClasses(
+        void* newValue, const AZ::TypeId& valueType) const
+    {
+        if (azrtti_typeid<SegmentationClassConfigList>() != valueType)
+        {
+            AZ_Assert(false, "Unexpected value type");
+            return AZ::Failure(AZStd::string("Unexpectedly received an invalid type as segmentation classes!"));
+        }
+
+        const auto& segmentationClasses = *reinterpret_cast<SegmentationClassConfigList*>(newValue);
+
+        bool unknownPresent, groundPresent;
+        unknownPresent = groundPresent = false;
+        for (const auto& segmentationClass : segmentationClasses)
+        {
+            if (segmentationClass.m_classId == SegmentationClassConfiguration::UnknownClass.m_classId &&
+                segmentationClass.m_className == SegmentationClassConfiguration::UnknownClass.m_className)
+            {
+                unknownPresent = true;
+            }
+
+            if (segmentationClass.m_classId == SegmentationClassConfiguration::GroundClass.m_classId &&
+                segmentationClass.m_className == SegmentationClassConfiguration::GroundClass.m_className)
+            {
+                groundPresent = true;
+            }
+        }
+
+        if (!unknownPresent || !groundPresent)
+        {
+            return AZ::Failure(
```

Comment:
While in practice this is good enough, I think this could return 3 state failure to also be able to signal missing both ground and unknown classes

Reply:
I'm not sure if this is not overkill, also this function is called after every change in the component's configuration. This would mean that it's impossible to remove two of these classes with one modification.